### PR TITLE
refactor payment processing classes

### PR DESCRIPTION
**Pull request by emma-green**

### CODE REVIEW

File: ecommerce/extensions/payment/views/cybersource.py [hunk 1]
```diff
@@ -176,15 +145,24 @@ def post(self, request):
         return super(CybersourceSubmitAPIView, self).post(request)
 
 
-class CybersourceNotificationMixin(CyberSourceProcessorMixin, OrderCreationMixin):
```
Comment (MatthewPiatetsky) [inline ecommerce/extensions/payment/views/cybersource.py:179]: why did this get removed?

**Review by MatthewPiatetsky — COMMENTED** [review]
could you clarify the goal of this PR? I assume it is to move the create_order code for both cybersource and paypal to the same place without changing the behavior?

**Review by cpennington — APPROVED** [review]
I was going to ask you to split each of the simple refactor steps into separate commits, and then I saw you had already done it! It made it much easier to review. :+1:

File: ecommerce/extensions/payment/views/cybersource.py (hunk 1 above)
Comment (emma-green) [reply ecommerce/extensions/payment/views/cybersource.py:179]: This was only inherited from in one place, and did not mirror the PayPal code. I moved the method inside of it.

**Comment by emma-green** [comment]
> could you clarify the goal of this PR? I assume it is to move the create_order code for both cybersource and paypal to the same place without changing the behavior?

The intent of this pr is to make it so that post-payment processing that concerns order creation is in as few places as possible. This will enable us manage risk when we significantly change it so that order creation happens earlier. This is just a re-factor pull request -- it should *not* have other impacts.

**Review by MatthewPiatetsky — APPROVED** [review]
if create_order is the method you'll later significantly change i'd consider writing a test for it, also we'd want to probably test an order locally or on stage before deploying

**Comment by emma-green** [comment]
I'll write a test for create_order when I change it. There is also a ticket to re-factor order creation testing that could cover that as well. Right now, successful order creation is tested even if this method is not explicitly tested.

**Comment by emma-green** [comment]
> I was going to ask you to split each of the simple refactor steps into separate commits, and then I saw you had already done it! It made it much easier to review. 👍
@cpennington Thoughts on whether I should squash before merging?

**Comment by cpennington** [comment]
I'd probably capitalize the sentences, but yes, I think it would be reasonable to not squash them.

**Review by inventhouse — APPROVED** [review]

**Review by julianajlk — APPROVED** [review]

**Pull request merged**